proto/internal/temporal/server/api/schedule/v1/message.proto

```diff
@@ -155,3 +158,59 @@ message HsmSchedulerState {
     google.protobuf.Timestamp next_invocation_time = 3;
 
 }
+
+// V2 Scheduler state
+message HsmSchedulerV2State {
+    temporal.server.api.enums.v1.Scheduler2State state = 1;
+
+    // scheduler request parameters and metadata. 
+    temporal.api.schedule.v1.Schedule schedule = 2;
+    temporal.api.schedule.v1.ScheduleInfo info = 3;
+    temporal.api.schedule.v1.SchedulePatch initial_patch = 4;
+
+    // state common to all generators is stored in the toplevel machine
+    string namespace = 5;
+    string namespace_id = 6;
+    string schedule_id = 7;
+
+    // Implemented as a sequence number. Useful for substate machines to
+    // invalidate transactions based on update requests
+    int64 conflict_token = 8;
+}
+
+// V2 Scheduler Generator state
+message GeneratorInternalState {
+    temporal.server.api.enums.v1.SchedulerGeneratorState state = 1;
+
+    // waiting state waits for the next_invocation_time before transitioning to buffering
+    google.protobuf.Timestamp next_invocation_time = 2;
+
+    // high water mark
+    google.protobuf.Timestamp last_processed_time = 3;
+
+    // Implemented as a sequence number. Useful for invalidating a stale
+    // Generator persisted state write.
+    int64 conflict_token = 4;
```

Comment:
I'm trying to avoid making an assumption that we can have transactions across multiple columns/documents in CHASM, since I don't think we have that in HSM today (with `MachineTransition` operating on a single item). If our framework lets me do a transaction across multiple state machines, we could use only the one in the top-level scheduler.

Reply:
MachineTransition is a single transition within a transaction on the entire tree. You can rely on that here. CHASM will have similar semantics.

Reply:
Got it - will update!